CRM/Mailing/Event/BAO/Unsubscribe.php

```diff
@@ -220,10 +220,10 @@ public static function &unsub_from_mailing($job_id, $queue_id, $hash, $return =
     // list.
 
     while (!empty($mailings)) {
-      $do->query("
+      $do = CRM_Core_DAO::executeQuery("
                 SELECT      $mg.entity_table as entity_table,
                             $mg.entity_id as entity_id
-                FROM        $mg
+                FROM        civicrm_mailing_group $mg
```

Comment:
@eileenmcnaughton this looks off to me, why are you specifiying the table name here?

Reply:
@seamuslee001 so as I understand it & @mlutfy can jump in if I am wrong we don't need to translate the table name - we just call CRM_Core_DAO::executeQuery and by default that has I10n handling on and it deals with it.

The reason THIS code has had to do all this mucking around is because it didn't want to conform & wanted to show independence by calling a retro function that doesn't handle I10n - but sometimes you just don't get to be an individual

Reply:
I think the other thing is what about when your coming via an extern URL? Like from the {mailing.opt_out_url} token. In those cases have we bootstrapped in all the CRM classes or just done require once?

Reply:
ah - good thinking - we should test!